src/Analysis/Ast/Impl/Specializations/Typing/TypingModule.cs

```diff
@@ -227,27 +235,32 @@ private IPythonType CreateTupleType(IReadOnlyList<IPythonType> typeArgs)
             return Interpreter.UnknownType;
         }
 
-        private IPythonType CreateTypeAlias(IArgumentSet args) {
-            var typeArgs = args.Values<IMember>();
-            if (typeArgs.Count == 2) {
-                var typeName = (typeArgs[0] as IPythonConstant)?.Value as string;
-                if (!string.IsNullOrEmpty(typeName)) {
-                    return new TypeAlias(typeName, typeArgs[1].GetPythonType() ?? Interpreter.UnknownType);
-                }
+        private IPythonType CreateTypeAlias(IArgumentSet argSet) {
+            if (argSet.Errors.Count > 0) {
+                argSet.ReportErrors();
+                return Interpreter.UnknownType;
+            }
+
+            var args = argSet.Values<IMember>();
+            var name = (args[0] as IPythonConstant)?.GetString();
 
-                var firstArgType = (typeArgs[0] as PythonInstance)?.Type.Name;
-                var eval = args.Eval;
-                var expression = args.Expression;
+            if (!string.IsNullOrEmpty(name)) {
+                return new TypeAlias(name, args[1].GetPythonType() ?? Interpreter.UnknownType);
+            }
 
+            // If first arg is not a string and not unknown, provide diagnostic
+            if (!args[0].IsUnknown() && args[0].GetPythonType() != Interpreter.GetBuiltinType(BuiltinTypeId.Str)) {
```

Comment:
`name` will be null (via `(args[0] as IPythonConstant)?.GetString();` and you already checked for non-null above, so it looks like this `if` is redundant.

Reply:
But the string could be empty, so I include that check specifically for that. `IsNullOrEmpty` also checks for empty string. 

Perhaps now I could change it to a regular null check.

Reply:
Yes, I mean if argument is not a string, then name is null. If name is empty, then it is a string.